meshtastic/interfaces/ble/lifecycle_ownership_runtime.py

```
@@ -0,0 +1,412 @@
+            not snapshot.still_owned
+            or snapshot.is_closing
+            or snapshot.lost_gate_ownership
+        ):
+            _raise_invalidated(snapshot)
+        prior_ever_connected = snapshot.prior_ever_connected
+
+        should_publish_connected = False
+        with iface._state_lock:
+            still_owned, is_closing = get_connected_status_locked(connected_client)
+            if still_owned and not is_closing:
+                if not iface._client_publish_pending:
```

Ruff (0.15.6)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)